src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/CopiedFieldRvaNode.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection.Metadata;
+using System.Reflection.Metadata.Ecma335;
+
+using Internal.Text;
+using Internal.TypeSystem;
+using Internal.TypeSystem.Ecma;
+
+using Debug = System.Diagnostics.Debug;
+
+namespace ILCompiler.DependencyAnalysis.ReadyToRun
+{
+    class CopiedFieldRvaNode : ObjectNode, ISymbolDefinitionNode
+    {
+        private EcmaField _field;
+
+        public CopiedFieldRvaNode(EcmaField field)
+        {
+            Debug.Assert(field.HasRva);
+
+            _field = (EcmaField)field.GetTypicalFieldDefinition();
+        }
+
+        public override ObjectNodeSection Section => ObjectNodeSection.TextSection;
```

Comment:
Technically, we should figure out what section the data is in and place it in the same one. IL allows .`text`, `.sdata`, and `.tls` (controlled by `.data cil`, `.data` and `.data tls` IL declarations, respectively).

`.text` is enough for C#. I fully expect to see `.sdata` for obfuscated assemblies at some point. `.tls` we hopefully won't have to deal with.

Maybe a TODO is enough for now though.

Reply:
I think it's more about the section characteristics than the name - if the input assembly wants the field in a writable section, the data shouldn't go into .text. Maybe that's all we need to do for that.

Reply:
OK, that sounds promising - I was originally thinking we would need to start putting RVA fields into the same exact section they were in originally; what you described sounds much easier.